### ISSUE TITLE
Reward and Punishment

### ISSUE THREAD

**Issue by FabianoLothor** [BrainJS/brain.js#825]
Hey guys, I didn't see this option in the docs.

Is there a way to reward or punish an input value?

**Comment by ashejuniper** [comment]
AFAIK, no, there isn't. With the Brain.js library, you just train on desired inputs and outputs, and the training algorithms "reward" good behavior (i.e. when the output matches the desired output for a given input), and the networks learn. I recommend looking at the examples, because most of them explain how training works in Brain.js.

(If you feel your question was not answered: Apologies in advance if you already know all this, but I recommend looking into how neural networks are trained, training algorithms, backpropigation, etc. since that'll help you achieve better-trained networks, which I _think_ is your goal here)

**Comment by robertleeplummerjr** [comment]
So the official brain.js repo doesn't yet support this.  BUT, I have been working some time on https://github.com/BrainJS/rl, which does.  It is a reinforcement ML library originally written by Andrej Karpathy, but I ported it to typescript and fixed a lot of issues, and consistencies.  It is stable, but it is actively improving.

**Comment by robertleeplummerjr** [comment]
Also, fyi, it doesn't yet have a "release", because I'm working with NPM to get the organization name corrected.  I'm working on releasing it as "@brain/rl".

**Comment by robertleeplummerjr** [comment]
Also, the documentation is off.  Read the typescript, not the docs.  I'll need to address the docs at some future point.

**Comment by FabianoLothor** [comment]
Hey guys, thank you for the answers.

I saw that TensorFlow provides this ability and I was wondering if we'd here on brain.js too.

@robertleeplummerjr I'm glad to know that you're adding this feature in that repository, I'll certainly take a look when I back to my project.

I'm closing this thread now, as a suggestion, it would be nice if we could add the reinforcement ability to brain.js too, it's a great project.

### Edit 

Just saw that the thread was already closed. :smile:
